### DISCUSSION TITLE
M1 Led is continuously Blinking Red

### DISCUSSION THREAD

**Discussion by Deeshant2234** [bitcraze/discussions#176] [Q&A]
When I am Connecting my motor with M1 it starts rotating immediately and M1 red led is continuously blinking.

**Comment by knmcguire** [comment]
Hi!

We need more information about your problem:
- Is this a completely new crazyflie or an old one?
- Is this a crazyflie with the latest release of the firmware, build from source or with any firmware modifications?
- Are any other decks attached or is it only the crazyflie control board?
- Are you able to connect to the crazyflie through the cfclient? Anything you can see from the console tab?

**Reply by stel227** [reply]
Hello there, I have the same problem. I was following https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/

- My CrazyFlie 2.1 is completely new
- Didn't try, because the very first test failed, it's factory setting
- Only the control board
- Didn't try, because the very first test failed

**Reply by krichardsson** [reply]
Hi @deaunforgiven 

It would be useful to get some information from the Crazyflie. You can skip mounting motors and so on and continue [at installing the client](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#inst-comp). When you have installed the client and connected to the Crazyflie (using the radio or USB), open the console log tab and send us a dump of the log. There should be some information in the log of what failed.

I was a bit short in the instructions above, let us know if you need further help to get the log :-)

**Reply by stel227** [reply]
Hello @krichardsson
Thanks for the quick reply!

So, I managed to connect to the flie via radio. I also flashed the latest firmware: 2022.09 cf2. cfclient was (just in case) reinstalled:
`python -m pip install --upgrade --force-reinstall cfclient`

Here's what I see when I run cfclient from cmd:

> UserWarning: Using SDL2 binaries from pysdl2-dll 2.24.2
> INFO:cfclient.gui:Disabling STL printouts
> INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
> INFO:cfclient.utils.input.inputreaders:Could not initialize [linuxjsdev]: Only supported on Linux
> INFO:cfclient.utils.input.inputreaders:Successfully initialized [pysdl2]
> INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
> INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
> INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
> INFO:cfclient.utils.config:Dist config read from C:\Users\budko\AppData\Local\Programs\Python\Python310\lib\site-packages\cfclient/configs/config.json
> INFO:cfclient.utils.config:Config file read from [C:\Users\budko\AppData\Local\Bitcraze\cfclient/config.json]
> INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file
> INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file
> INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file
> INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]
> WARNING:cfclient.ui.main:failed to parse address from config: invalid literal for int() with base 16: '2M'
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cflib.crtp.radiodriver:v0.53 dongle with serial N/A found
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tabs.QualisysTab:Switching Flight Mode to: FlightModeStates.DISCONNECTED
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
> INFO:cfclient.ui.tab_toolbox:No config found for open_toolboxes
> INFO:cfclient.ui.tab_toolbox:No config found for open_toolboxes
> INFO:cfclient.utils.input.inputreaders.pysdl2:Looking for devices
> INFO:cfclient.utils.input.inputreaders.pysdl2:Found 1 devices
> INFO:cfclient.utils.input:Selected MUX: Normal
> INFO:cfclient.ui.main:Role of Nintendo Switch Pro Controller is Device
> INFO:cfclient.utils.input.mux:Adding device Nintendo Switch Pro Controller to MUX Normal
> INFO:cfclient.ui.tabs.QualisysTab:Exception during qtm discovery: 'QRTDiscoveryProtocol' object has no attribute 'error_received'
> INFO:cflib.drivers.cfusb:Looking for devices....
> WARNING:cflib.crtp.usbdriver:Exception while scanning for Crazyflie USB: 'NoneType' object has no attribute 'bcdDevice'

At this moment the GUI is launched and I click "Connect":

> INFO:cflib.crazyflie:Callback->Connection initialized[radio://0/80/2M]
> INFO:cflib.crazyflie:We are connected[radio://0/80/2M], request connection setup
> INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
> INFO:cflib.crazyflie:Callback->Connected to [radio://0/80/2M]
> INFO:cflib.crazyflie.platformservice:_crt_service_callback
> INFO:cflib.crazyflie.platformservice:Request protocol version
> INFO:cflib.crazyflie.platformservice:_platform_callback
> INFO:cflib.crazyflie.platformservice:Protocol version (platform): 5
> INFO:cflib.crazyflie.toccache:Saved cache to [C:\Users\budko\AppData\Local\Bitcraze\cfclient/cache/644160F3.json]
> INFO:cflib.crazyflie:Log TOC finished updating
> INFO:cflib.crazyflie.mem:4 memories found
> INFO:cflib.crazyflie:Memories finished updating
> INFO:cflib.crazyflie.toccache:Saved cache to [C:\Users\budko\AppData\Local\Bitcraze\cfclient/cache/E9DA3617.json]
> INFO:cflib.crazyflie:Param TOC finished updating
> INFO:cflib.crazyflie:Callback->Connection setup finished [radio://0/80/2M]
> INFO:cfclient.utils.logconfigreader:Parsing [stabilizer.json]
> WARNING: QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
> (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
> INFO:cflib.crazyflie.log:Have successfully started logging for id=1
> INFO:cflib.crazyflie.log:Have successfully started logging for id=3
> INFO:cfclient.ui.tabs.FlightTab:[imu_sensors.BMP388]: 1
> INFO:cfclient.ui.tabs.FlightTab:[imu_sensors.AK8963]: 0
> INFO:cfclient.ui.tabs.FlightTab:[imu_sensors.LPS25H]: 0
> INFO:cflib.crazyflie:All parameters updated
> INFO:cflib.crazyflie:Callback->Connection completed [radio://0/80/2M]
> WARNING: QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING: QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> INFO:cflib.crazyflie.log:Have successfully started logging for id=4
> INFO:cfclient.ui.tabs.locopositioning_tab:Crazyflie connected to radio://0/80/2M
> INFO:cfclient.ui.tabs.locopositioning_tab:Requesting loco deck parameter
> WARNING: QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> INFO:cfclient.ui.tabs.lighthouse_tab:Crazyflie connected to radio://0/80/2M
> WARNING: QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> WARNING:vispy:QObject::connect: Cannot queue arguments of type 'QVector<int>'
> (Make sure 'QVector<int>' is registered using qRegisterMetaType().)
> INFO:cfclient.ui.tabs.FlightTab:Changed effect to 0
> INFO:cfclient.ui.tabs.locopositioning_tab:No Loco deck installed
> INFO:cfclient.ui.tabs.FlightTab:[imu_sensors.BMP388]: 1
> INFO:cfclient.ui.tabs.FlightTab:[imu_sensors.AK8963]: 0
> INFO:cfclient.ui.tabs.FlightTab:[imu_sensors.LPS25H]: 0
> INFO:cfclient.ui.tabs.FlightTab:Changed effect to 0

I hope this clarifies the issue. Btw, I see the PFD responding as I change drone's attitude (by moving it with my hand).

**Reply by krichardsson** [reply]
Great, thanks!

What I actually was looking for is the [log in the console log tab in the client](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/console_tab/), sorry this was not obvious from my short description. Now that you have done all the work of setting up the client it should be easy get that as well :-) You can enable it in the view/tabs/Console menu.

**Reply by stel227** [reply]
Here you go:

> SYS: Crazyflie 2.1 is up and running!
> SYS: Build 0:5c4261e5a9ff (2022.09) CLEAN
> SYS: I am 0x2032353241335019002D003C and I have 1024KB of flash!
> CFGBLK: v1, verification [OK]
> DECK_CORE: 0 deck(s) found
> IMU: BMI088: Using I2C interface.
> IMU: BMI088 Gyro connection [OK].
> IMU: BMI088 Accel connection [OK]
> IMU: BMP388 I2C connection [OK]
> ESTIMATOR: Using Complementary (1) estimator
> CONTROLLER: Using PID (1) controller
> MTR-DRV: Using brushed motor driver
> SYS: About to run tests in system.c.
> SYS: NRF51 version: 2022.09
> EEPROM: I2C connection [OK].
> STORAGE: Storage check [OK].
> IMU: BMI088 gyro self-test [OK]
> SYS: Self test passed!
> STAB: Wait for sensor calibration...
> SYS: Free heap: 20496 bytes
> STAB: Ready to fly.

**Reply by krichardsson** [reply]
I think this is good news! From the log it looks like your Crazyflie works as expected, if there was a problem it would say something like "Self test failed". 

When everything is fine, the M1 LED is also blinking red, twice a second in a steady rate. Is this perhaps what you see? (We know it is slightly confusing and green would maybe have been a better choise, but it is due to historical reasons)

If the self test fails you will get [bursts of five blinks like in the video in the tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#self-test-fails).

**Reply by stel227** [reply]
Ah, one should read a few screens lower:
```
Understanding LEDs
You also need to understand what the LEDs mean.

Power on and all is good: The blue LEDs (2 and 3) are fully lit and the front right LED (1) is blinking red twice every second.
```

Maybe "Test" UM chapter could be extended by adding (in addition to "self-test" & "self-test fails") "self test passed" subchapter which would state explicitly that 2Hz M1 RED blinking is healthy. "self-test" ch. says "If the M4 LED blinks GREEN five times fast, then the test has passed." But maybe it could say smth like: "If the M4 LED blinks GREEN five times fast and then M1 LED blinks RED at 2Hz frequency, then the test has passed."

Anyways, I'm glad that the system works. Thanks for the support!

**Reply by krichardsson** [reply]
Great!

Yeah, adding a video of what it should look like is probably a good idea :-) 
Have fun with the Crazyflie!
